src/index.js

```diff
@@ -43,9 +44,15 @@ class Libp2p extends EventEmitter {
     // and add default values where appropriate
     this._options = validateConfig(_options)
 
-    this.datastore = this._options.datastore
     this.peerId = this._options.peerId
-    this.peerStore = new PeerStore()
+    this.datastore = this._options.datastore
+
+    this.peerStore = !(this.datastore && this._options.peerStore.persistence)
+      ? new PeerStore()
+      : new PersistentPeerStore({
+        datastore: this.datastore,
+        ...this._options.peerStore
+      })
```

Comment:
nit: flip this to avoid inverting the conditional, it's harder to follow that way

```suggestion
    this.peerStore = (this.datastore && this._options.peerStore.persistence)
      ? new PersistentPeerStore({
        datastore: this.datastore,
        ...this._options.peerStore
      })
      : new PeerStore()
```